wisdem/glue_code/gc_WT_InitModel.py

```diff
@@ -601,6 +601,80 @@ def assign_internal_structure_2d_fem_values(wt_opt, modeling_options, internal_s
 
     return wt_opt
 
+def assign_user_elastic(wt_opt, modeling_options, user_elastic_properties):
+    # Function to assign values to the openmdao component Blade_Internal_Structure_2D_FEM
+    n_span = modeling_options["WISDEM"]["RotorSE"]["n_span"]
+    nd_span = wt_opt["blade.outer_shape_bem.s_default"]
+    # TODO YL: maybe I can pass in the inertia twist throught the twist in six_x_six
+    stiff_grid = user_elastic_properties["six_x_six"]["stiff_matrix"]["grid"]
+    stiff_matrix = np.array(user_elastic_properties["six_x_six"]["stiff_matrix"]["values"])
+
+    inertia_grid = user_elastic_properties["six_x_six"]["inertia_matrix"]["grid"]
+    inertia_matrix = np.array(user_elastic_properties["six_x_six"]["inertia_matrix"]["values"])
+
+    # 21-element inertia matrix
+    # idx = [0, 1, 2, 3, 4, 5,     6, 7, 8, 9, 10,   11, 12,   13,    14, 15,    16,   17, 18,    19, 20]
+    # M   = [m, 0, 0, 0, 0, -mYcm, m, 0, 0, 0, mXcm, m,  mYcm, -mXcm, 0,  iedge, -icp, 0,  iflap, 0,  iplr]
+
+    # 21-element stiffness matrix
+    # idx = [0,        1, 2, 3, 4, 5, 6,        7, 8, 9, 10, 11, 12, 13, 14, 15,     16, 17, 18,     19, 20]
+    # K =   [KShrflap, 0, 0, 0, 0, 0, KShredge, 0, 0, 0, 0,  EA, 0,  0,  0,  EIedge, 0,  0,  EIflap, 0,  GJ]
+
+    # Assemble stiffnees and inertia matrices
+    K11 = PchipInterpolator(stiff_grid, stiff_matrix["K11"][:])(nd_span)
+    K22 = PchipInterpolator(stiff_grid, stiff_matrix["K22"][:])(nd_span)
+    K33 = PchipInterpolator(stiff_grid, stiff_matrix["K33"][:])(nd_span)
+    K44 = PchipInterpolator(stiff_grid, stiff_matrix["K44"][:])(nd_span)
+    K55 = PchipInterpolator(stiff_grid, stiff_matrix["K55"][:])(nd_span)
+    K66 = PchipInterpolator(stiff_grid, stiff_matrix["K66"][:])(nd_span)
+    K12 = PchipInterpolator(stiff_grid, stiff_matrix["K12"][:])(nd_span)
+    K13 = PchipInterpolator(stiff_grid, stiff_matrix["K13"][:])(nd_span)
+    K14 = PchipInterpolator(stiff_grid, stiff_matrix["K14"][:])(nd_span)
+    K15 = PchipInterpolator(stiff_grid, stiff_matrix["K15"][:])(nd_span)
+    K16 = PchipInterpolator(stiff_grid, stiff_matrix["K16"][:])(nd_span)
+    K23 = PchipInterpolator(stiff_grid, stiff_matrix["K23"][:])(nd_span)
+    K24 = PchipInterpolator(stiff_grid, stiff_matrix["K24"][:])(nd_span)
+    K25 = PchipInterpolator(stiff_grid, stiff_matrix["K25"][:])(nd_span)
+    K26 = PchipInterpolator(stiff_grid, stiff_matrix["K26"][:])(nd_span)
+    K34 = PchipInterpolator(stiff_grid, stiff_matrix["K34"][:])(nd_span)
+    K35 = PchipInterpolator(stiff_grid, stiff_matrix["K35"][:])(nd_span)
+    K36 = PchipInterpolator(stiff_grid, stiff_matrix["K36"][:])(nd_span)
+    K45 = PchipInterpolator(stiff_grid, stiff_matrix["K45"][:])(nd_span)
+    K46 = PchipInterpolator(stiff_grid, stiff_matrix["K46"][:])(nd_span)
+    K56 = PchipInterpolator(stiff_grid, stiff_matrix["K56"][:])(nd_span)
+
+    wt_opt["rotorse.EA"] = K33
```

Comment:
formally this is not correct (I think...). the properties of PreComp are defined according to the user manual, whereas windIO defines properties with respect to the reference axis. indeed when we go from pre comp to windIo we have to rototranslate things, and formally here we should be taking the opposite steps